comms/dht/src/crypt.rs

```diff
@@ -63,10 +89,10 @@ mod test {
     fn decrypt_fn() {
         let key = CommsPublicKey::default();
         let cipher_text =
-            from_hex("7ecafb4c0a88325c984517fca1c529b3083e9976290a50c43ff90b2ccb361aeaabfaf680e744b96fc3649a447b")
+            from_hex("24bf9e698e14938e93c09e432274af7c143f8fb831f344f244ef02ca78a07ddc28b46fec536a0ca5c04737a604")
```

Comment:
This is odd, I thought only edge cases were broken on the handrolled chacha. If so, then this should still work with the same input and this suggests that the API has changed significantly, meaning other parts of comms could break.

Reply:
So looked further into this and I cannot actually get the RustCrypto output to agree with the custom implementation or with the test vectors defined in https://datatracker.ietf.org/doc/html/rfc8439#section-2.4.1. However, I checked the output against the Python Cryptodome library and the RustCrypto and Cryptodome implementations agree 🤷‍♂️